packages/foam-vscode/src/features/backlinks.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { commands, workspace, window } from 'vscode';
+import { URI, FoamWorkspace, IDataStore, Resource, Note } from 'foam-core';
+import {
+  cleanWorkspace,
+  closeEditors,
+  createNote,
+  createTestNote,
+} from '../test/test-utils';
+import { BacklinksTreeDataProvider, BacklinkTreeItem } from './backlinks';
+import { ResourceTreeItem } from '../utils/grouped-resources-tree-data-provider';
+
+describe('Backlinks panel', () => {
+  beforeAll(async () => {
+    await cleanWorkspace();
+    await createNote(noteA);
+    await createNote(noteB);
+    await createNote(noteC);
+  });
+  afterAll(cleanWorkspace);
+
+  const rootUri = workspace.workspaceFolders[0].uri;
+  const ws = new FoamWorkspace();
+  const dataStore = {
+    read: uri => {
+      return Promise.resolve('');
+    },
+    isMatch: uri => uri.path.endsWith('.md'),
+  } as IDataStore;
+
+  const noteA = createTestNote({
+    root: rootUri,
+    uri: './note-a.md',
+  });
+  const noteB = createTestNote({
+    root: rootUri,
+    uri: './note-b.md',
+    links: [{ slug: 'note-a' }, { slug: 'note-a' }],
+  });
+  const noteC = createTestNote({
+    root: rootUri,
+    uri: './note-c.md',
+    links: [{ slug: 'note-a' }],
+  });
+  ws.set(noteA)
+    .set(noteB)
+    .set(noteC)
+    .resolveLinks(true);
+
+  const provider = new BacklinksTreeDataProvider(ws, dataStore);
+
+  beforeEach(async () => {
+    await closeEditors();
+    provider.target = undefined;
+  });
+
+  // Skipping these as still figuring out how to interact with the provider
```

Comment:
still figuring out how to do this. I kept the tests because this is part of the spec, but of course they don't work now because the `provider` instance I am talking to is not the one used by VS Code (or at least I believe this is why the "targets active editor" test fails)

Reply:
In case this jogs a brainwave (probably completely wrong!) - could this be solved in vscode-jest-environment (or closely similar name)?